### DISCUSSION TITLE
How can Kong implement cluster mode for precise concurrent rate limiting?

### DISCUSSION THREAD

**Discussion by xlszht** [Kong/kong#13660] [Help]
I have set up rate limiting to 5 requests per second, but with 20 threads making requests simultaneously, I find that most of the background requests are close to 20 times per second, which doesn't seem to limit the rate accurately.
How can I control the rate more precisely, such as setting a limit of 5 requests per second, while actually accepting up to 7 requests per second as acceptable?

**Comment by chobits** [comment]
For your question in your discussion title, I guess you want to know if kong rate limiting plugin supports cluster. See this documentation: https://docs.konghq.com/hub/kong-inc/rate-limiting/#strategies, you could use `redis` strategy to enable cluster mode.

**Reply by xlszht** [reply]
thx,i mean i use PostgreSQL cluster mode，but not support concurrent scenario, because the implementation principle is a counter,  unable to handle traffic spikes. I want to find some methods to make rate limiting more effective.
